[Bug]: Go integration tests fail on macOS

### Steps to reproduce

Run the `build.yml` workflow with Go integration tests enabled.

### Actual behaviour

MacOS tests fail:
https://github.com/dstackai/dstack/actions/runs/17090335979/job/48462629999

```
go version go1.23.8 darwin/arm64
?   	github.com/dstackai/dstack/runner/cmd/runner	[no test files]
?   	github.com/dstackai/dstack/runner/cmd/shim	[no test files]
?   	github.com/dstackai/dstack/runner/consts	[no test files]
?   	github.com/dstackai/dstack/runner/internal/api	[no test files]
ok  	github.com/dstackai/dstack/runner/internal/common	1.016s
ok  	github.com/dstackai/dstack/runner/internal/connections	1.205s
?   	github.com/dstackai/dstack/runner/internal/gerrors	[no test files]
?   	github.com/dstackai/dstack/runner/internal/log	[no test files]
?   	github.com/dstackai/dstack/runner/internal/runner/api	[no test files]
?   	github.com/dstackai/dstack/runner/internal/schemas	[no test files]
?   	github.com/dstackai/dstack/runner/internal/shim/backends	[no test files]
?   	github.com/dstackai/dstack/runner/internal/types	[no test files]
ok  	github.com/dstackai/dstack/runner/internal/executor	4.675s
ok  	github.com/dstackai/dstack/runner/internal/metrics	1.090s
ok  	github.com/dstackai/dstack/runner/internal/repo	1.104s
time=2025-08-20T06:24:29.987214Z level=info msg=skip locking: GPU already locked id=GPU-beef
time=2025-08-20T06:24:29.987353Z level=warning msg=skip locking: unknown GPU resource id=GPU-dead
time=2025-08-20T06:24:29.987455Z level=warning msg=skip releasing: unknown GPU resource id=GPU-dead
time=2025-08-20T06:24:29.987473Z level=info msg=skip releasing: GPU not locked id=GPU-c0de
--- FAIL: TestDocker_SSHServer (0.01s)
panic: runtime error: invalid memory address or nil pointer dereference
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x100a66634]

goroutine 34 [running]:
testing.tRunner.func1.2({0x100bb2ba0, 0x100ef9850})
	/Users/runner/hostedtoolcache/go/1.23.8/arm64/src/testing/testing.go:1632 +0x2c4
testing.tRunner.func1()
	/Users/runner/hostedtoolcache/go/1.23.8/arm64/src/testing/testing.go:1635 +0x47c
panic({0x100bb2ba0?, 0x100ef9850?})
	/Users/runner/hostedtoolcache/go/1.23.8/arm64/src/runtime/panic.go:791 +0x124
github.com/dstackai/dstack/runner/internal/shim.(*DockerRunner).Remove(0x0, {0x100c43e90, 0x100f40f40}, {0xc000188110, 0x10})
	/Users/runner/work/dstack/dstack/runner/internal/shim/docker.go:454 +0x64
panic({0x100bb2ba0?, 0x100ef9850?})
	/Users/runner/hostedtoolcache/go/1.23.8/arm64/src/runtime/panic.go:791 +0x124
github.com/dstackai/dstack/runner/internal/shim.(*DockerRunner).Submit(_, {_, _}, {{0xc000188110, 0x10}, {0x100a83830, 0x14}, {0x0, 0x0}, {0x0, ...}, ...})
	/Users/runner/work/dstack/dstack/runner/internal/shim/docker.go:242 +0x120
github.com/dstackai/dstack/runner/internal/shim.TestDocker_SSHServer(0xc00031e1a0)
	/Users/runner/work/dstack/dstack/runner/internal/shim/docker_test.go:42 +0x3d4
testing.tRunner(0xc00031e1a0, 0x100c3bd18)
	/Users/runner/hostedtoolcache/go/1.23.8/arm64/src/testing/testing.go:1690 +0x188
created by testing.(*T).Run in goroutine 1
	/Users/runner/hostedtoolcache/go/1.23.8/arm64/src/testing/testing.go:1743 +0x5e4
FAIL	github.com/dstackai/dstack/runner/internal/shim	0.078s
ok  	github.com/dstackai/dstack/runner/internal/shim/api	1.066s
ok  	github.com/dstackai/dstack/runner/internal/shim/dcgm	1.085s
ok  	github.com/dstackai/dstack/runner/internal/shim/host	1.102s
FAIL
```

### Expected behaviour

Tests don't fail.

### dstack version

0.19.24

### Server logs

```shell

```

### Additional information

_No response_

--- a/runner/internal/shim/docker_test.go
+++ b/runner/internal/shim/docker_test.go
@@ -6,6 +6,7 @@ import (
 	"math/rand"
 	"os"
 	"os/exec"
+	"runtime"
 	"strconv"
 	"strings"
 	"sync"
@@ -35,7 +36,9 @@ func TestDocker_SSHServer(t *testing.T) {
 	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(timeout)*time.Second)
 	defer cancel()
 
-	dockerRunner, _ := NewDockerRunner(ctx, params)
+	dockerRunner, err := NewDockerRunner(ctx, params)
+	require.NoError(t, err)
+
 	taskConfig := createTaskConfig(t)
 	defer dockerRunner.Remove(context.Background(), taskConfig.ID)
 
@@ -45,7 +48,7 @@ func TestDocker_SSHServer(t *testing.T) {
 
 // TestDocker_SSHServerConnect pulls ubuntu image (without sshd), installs openssh-server and tries to connect via SSH
 func TestDocker_SSHServerConnect(t *testing.T) {
-	if testing.Short() {
+	if testing.Short() || (os.Getenv("CI") == "true" && runtime.GOOS == "darwin") {
 		t.Skip()
 	}
 	t.Parallel()
@@ -66,7 +69,8 @@ func TestDocker_SSHServerConnect(t *testing.T) {
 	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(timeout)*time.Second)
 	defer cancel()
 
-	dockerRunner, _ := NewDockerRunner(ctx, params)
+	dockerRunner, err := NewDockerRunner(ctx, params)
+	require.NoError(t, err)
 
 	var wg sync.WaitGroup
 	wg.Add(1)
@@ -99,7 +103,7 @@ func TestDocker_SSHServerConnect(t *testing.T) {
 }
 
 func TestDocker_ShmNoexecByDefault(t *testing.T) {
-	if testing.Short() {
+	if testing.Short() || (os.Getenv("CI") == "true" && runtime.GOOS == "darwin") {
 		t.Skip()
 	}
 	t.Parallel()
@@ -113,7 +117,9 @@ func TestDocker_ShmNoexecByDefault(t *testing.T) {
 	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(timeout)*time.Second)
 	defer cancel()
 
-	dockerRunner, _ := NewDockerRunner(ctx, params)
+	dockerRunner, err := NewDockerRunner(ctx, params)
+	require.NoError(t, err)
+
 	taskConfig := createTaskConfig(t)
 	defer dockerRunner.Remove(context.Background(), taskConfig.ID)
 
@@ -122,7 +128,7 @@ func TestDocker_ShmNoexecByDefault(t *testing.T) {
 }
 
 func TestDocker_ShmExecIfSizeSpecified(t *testing.T) {
-	if testing.Short() {
+	if testing.Short() || (os.Getenv("CI") == "true" && runtime.GOOS == "darwin") {
 		t.Skip()
 	}
 	t.Parallel()
@@ -136,7 +142,9 @@ func TestDocker_ShmExecIfSizeSpecified(t *testing.T) {
 	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(timeout)*time.Second)
 	defer cancel()
 
-	dockerRunner, _ := NewDockerRunner(ctx, params)
+	dockerRunner, err := NewDockerRunner(ctx, params)
+	require.NoError(t, err)
+
 	taskConfig := createTaskConfig(t)
 	taskConfig.ShmSize = 1024 * 1024
 	defer dockerRunner.Remove(context.Background(), taskConfig.ID)

--- a/runner/internal/shim/docker_test.go
+++ b/runner/internal/shim/docker_test.go
@@ -21,7 +21,7 @@ import (
 
 // TestDocker_SSHServer pulls ubuntu image (without sshd), installs openssh-server and exits
 func TestDocker_SSHServer(t *testing.T) {
-	if testing.Short() {
+	if testing.Short() || (os.Getenv("CI") == "true" && runtime.GOOS == "darwin") {
 		t.Skip()
 	}
 	t.Parallel()

